shell/common/color_util.cc

```diff
@@ -6,44 +6,52 @@
 
 #include <vector>
 
-#include "base/strings/string_number_conversions.h"
 #include "base/strings/string_util.h"
 #include "base/strings/stringprintf.h"
+#include "content/public/common/color_parser.h"
+#include "ui/gfx/color_utils.h"
 
 namespace electron {
 
-SkColor ParseHexColor(const std::string& color_string) {
-  // Check the string for incorrect formatting.
-  if (color_string.empty() || color_string[0] != '#')
-    return SK_ColorWHITE;
+std::string SkColorToColorString(SkColor color, const std::string& format) {
+  double alpha_double = (double)(SkColorGetA(color)) / 255.0;
+  double alpha =
+      alpha_double <= 0.0 ? 0.0 : alpha_double >= 1.0 ? 1.0 : alpha_double;
 
-  // Prepend FF if alpha channel is not specified.
-  std::string source = color_string.substr(1);
-  if (source.size() == 3)
-    source.insert(0, "F");
-  else if (source.size() == 6)
-    source.insert(0, "FF");
+  if (format == "hsl" || format == "hsla") {
+    color_utils::HSL hsl;
 
-  // Convert the string from #FFF format to #FFFFFF format.
-  std::string formatted_color;
-  if (source.size() == 4) {
-    for (size_t i = 0; i < 4; ++i) {
-      formatted_color += source[i];
-      formatted_color += source[i];
+    // Hue ranges between 0 - 360, and saturation/lightness both range from 0 -
+    // 100%, so multiply appropriately to convert to correct int values.
+    color_utils::SkColorToHSL(color, &hsl);
+    if (format == "hsla") {
+      return base::StringPrintf("hsl(%ld, %ld%%, %ld%%, %.1f)",
+                                lround(hsl.h * 360), lround(hsl.s * 100),
+                                lround(hsl.l * 100), alpha);
```

Comment:
I might be reading `ui/gfx/color_utils.cc` wrong but it looks like the range for HSL.l is [0...255], not [0..100]?